src/create.cpp

```diff
@@ -165,14 +165,23 @@ static void playerClearHistory() {
     }
 }
 
+static int16_t calculateSocialClass (int16_t social_class) {
+    if (social_class > 100) {
+        return 100;
+    } else if (social_class < 1) {
+        return 1;
+    }
+    return social_class;
+}
+
 // Get the racial history, determines social class -RAK-
 //
 // Assumptions:
 //   - Each race has init history beginning at (race-1)*3+1
 //   - All history parts are in ascending order
 static void characterGetHistory() {
     int history_id = py.misc.race_id * 3 + 1;
-    int social_class = randomNumber(4);
+    int16_t social_class = calculateSocialClass((int16_t) randomNumber(4));
```

Comment:
You will calculate the class incorrectly by calling the function here, as `social_class` is first adjusted on line `205`, inside the `while` loop.

The correct place to call it is _likely_ when updating the player (please double check my assumption)

`py.misc.social_class = calculateSocialClass(social_class);`